### PR TITLE
documentation(EJ2-67088): Added UG for lazy loading and load child ondemand with virtualization in remote data binding

### PR DESCRIPTION
**Pull request by Priyadarshan97**
Added UG for lazy loading and load child ondemand with virtualization in remote data binding.

### CODE REVIEW

File: ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md [hunk 1]
```diff
@@ -196,4 +196,70 @@ The argument passed to the [`ActionFailure`](https://help.syncfusion.com/cr/cref
 
 
 > The [`ActionFailure`](https://help.syncfusion.com/cr/cref_files/aspnetcore-js2/Syncfusion.EJ2~Syncfusion.EJ2.TreeGrid.TreeGrid~ActionFailure.html) event will be triggered not only for the server errors, but also when there is an exception while processing the treegrid actions.
-> You can refer to our [`ASP.NET MVC Tree Grid`](https://www.syncfusion.com/aspnet-mvc-ui-controls/tree-grid) feature tour page for its groundbreaking feature representations. You can also explore our [`ASP.NET MVC Tree Grid example`](https://ej2.syncfusion.com/aspnetmvc/TreeGrid/Overview#/material) to knows how to present and manipulate data.
\ No newline at end of file
+> You can refer to our [`ASP.NET MVC Tree Grid`](https://www.syncfusion.com/aspnet-mvc-ui-controls/tree-grid) feature tour page for its groundbreaking feature representations. You can also explore our [`ASP.NET MVC Tree Grid example`](https://ej2.syncfusion.com/aspnetmvc/TreeGrid/Overview#/material) to knows how to present and manipulate data.
+
+
+
+## Lazy loading with virtualization
```
Comment (alansangeeths) [inline ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md:203]: Change title to "Load on demand with virtualization"

File: ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md [hunk 2]
```diff
@@ -196,4 +196,70 @@ The argument passed to the [`ActionFailure`](https://help.syncfusion.com/cr/cref
...
\ No newline at end of file
+> You can refer to our [`ASP.NET MVC Tree Grid`](https://www.syncfusion.com/aspnet-mvc-ui-controls/tree-grid) feature tour page for its groundbreaking feature representations. You can also explore our [`ASP.NET MVC Tree Grid example`](https://ej2.syncfusion.com/aspnetmvc/TreeGrid/Overview#/material) to knows how to present and manipulate data.
+
+
+
+## Lazy loading with virtualization
+
+While binding remote data to Tree Grid component, by default Tree Grid renders parent rows in collapsed state. When expanding the root node, the child nodes will be loaded from the remote server.
+
+When using virtualization with remote data binding, it helps you to improve the tree grid performance while loading a large set of remote data by setting [`EnableVirtualization`](https://help.syncfusion.com/cr/aspnetmvc-js2/Syncfusion.EJ2.TreeGrid.TreeGrid.html#Syncfusion_EJ2_TreeGrid_TreeGrid_EnableVirtualization) as true. The Tree Grid UI virtualization allows it to render only rows and columns visible within the view-port without buffering the entire datasource.
```
Comment (alansangeeths) [inline ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md:207]: while loading a large set of data by setting

File: ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md [hunk 3]
```diff
@@ -196,4 +196,70 @@ The argument passed to the [`ActionFailure`](https://help.syncfusion.com/cr/cref
...
+{% endtabs %}
+{% endif %}
+
+
+
+> The [`ExpandStateMapping`](https://help.syncfusion.com/cr/aspnetmvc-js2/syncfusion.ej2.treegrid.treegrid.html#Syncfusion_EJ2_TreeGrid_TreeGrid_ExpandStateMapping) property maps the field name in data source, that denotes whether parent record is in expanded or collapsed state and this is useful to maintain the parent record state in server end.
+
+
+
+## LoadChildOnDemand with virtualization
```
Comment (alansangeeths) [inline ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md:236]: Move it to sub heading with ###
Change title to "Load parent rows in expanded state with virtualization".

File: ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md [hunk 4]
```diff
@@ -196,4 +196,70 @@ The argument passed to the [`ActionFailure`](https://help.syncfusion.com/cr/cref
...
+{% endhighlight %}
+{% highlight c# tabtitle="LoadChildOnDemand.cs" %}
+{% include code-snippet/tree-grid/data-binding-mvc/load-child-ondemand-virtualization/loadChildOnDemand.cs %}
+{% endhighlight %}
+{% endtabs %}
+{% endif %}
+
+
+
+> The [`ExpandStateMapping`](https://help.syncfusion.com/cr/aspnetmvc-js2/syncfusion.ej2.treegrid.treegrid.html#Syncfusion_EJ2_TreeGrid_TreeGrid_ExpandStateMapping) property maps the field name in data source, that denotes whether parent record is in expanded or collapsed state and this is useful to maintain the parent record state in server end.
```
Comment (alansangeeths) [inline ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md:265]: No need to mention this under load on demand data binding. Move it under common data binding topic https://ej2.syncfusion.com/documentation/treegrid/data-binding/data-binding/

File: ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md (hunk 2 above)
Comment (alansangeeths) [inline ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md:207]: Give a one line information about HasCHildMapping API also

File: ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md (hunk 1 above)
Comment (Priyadarshan97) [reply ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md]: updated

File: ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md (hunk 2 above)
Comment (Priyadarshan97) [reply ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md]: updated

File: ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md (hunk 4 above)
Comment (Priyadarshan97) [reply ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md]: updated

File: ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md (hunk 2 above)
Comment (Priyadarshan97) [reply ej2-asp-core-mvc/tree-grid/EJ2_ASP.MVC/data-binding/remote-data.md]: updated

**Review by thavasianand — APPROVED** [review]
changes are fine

**Pull request merged**